test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,98 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+            string dataPath = "iris-data.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, ctx => (
+               Label: ctx.LoadText(0),
+               SepalWidth: ctx.LoadFloat(1),
+               SepalLength: ctx.LoadFloat(2),
+               PetalWidth: ctx.LoadFloat(3),
+               PetalLength: ctx.LoadFloat(4)),
+               dataPath);
+
+            // Load the data into the system.
+            var data = dataReader.Read(dataPath);
+
+            var preprocess = data.Schema.MakeEstimator(row => (
+                // Convert string label to key.
+                Label: row.Label.Dictionarize(),
+                // Concatenate all features into a vector.
+                Features: row.SepalWidth.ConcatWith(row.SepalLength, row.PetalWidth, row.PetalLength)));
+
+
+            var pipeline = preprocess
+                .Append(row => row.Label.PredictWithDecisionTrees(row.Features))
+                // shoul it be BagVectorize instead of KeyToValue?
+                .Append(row => row.PredictedLabel.KeyToValue());
+
+            var model = pipeline.Fit(data);
+
+            var predictions = model.Transform(dataReader.Read(testDataPath));
+            var evaluator = new MultiClassEvaluator(env);
+            var metrics = evaluator.Evaluate(predictions);
+        }
+
+        [Fact]
+        public void TwitterSentimentAnalysis()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+            var dataPath = "wikipedia-detox-250-line-data.tsv";
+            // Load the data into the system.
+            var data = TextLoader.CreateReader(env, ctx => (
+                   Label: ctx.LoadFloat(0),
+                   Text: ctx.LoadText(1)),
+                   dataPath, hasHeader: true).Read(dataPath);
+
+            var preprocess = data.Schema.MakeEstimator(row => (
+                Label: row.Label,
+                // Concatenate all features into a vector.
+                Features: row.Text.TextFeaturizer()));
+
+            var pipeline = preprocess.
+                Append(row => row.Label.TrainLinearClassification(row.Features));
+
+            var (trainData, testData) = CrossValidator.TrainTestSplit(env, data: data, trainFraction: 0.7);
```

Comment:
>var (trainData, testData) = CrossValidator.TrainTestSplit(env, data: data, trainFraction: 0.7); [](start = 12, length = 95)

Let's have Tom comment: does this sound like it'll break us out of pigsty? So `pipeline.Fit(trainData)` is not possible? Or should we have `CrossValidator.TrainTestSplit` be pigsty-capable?

Reply:
Hi @Zruty0 , if we view it as desirable to have a method with signature `(DataView<TShape> split1, DataView<TShape> split2) Split(IDataView<TShape> source, double ratio)`, assuredly nothing prevents us from doing so. But that's fairly obviously the case, so I wonder if I've missed the point of the question somehow?